src/Examine.Core/Search/FacetLongField.cs

```diff
@@ -0,0 +1,19 @@
+using Lucene.Net.Facet.Range;
+
+namespace Examine.Search
+{
+    public class FacetLongField : IFacetLongField
+    {
+        public string Field { get; }
+
+        public Int64Range[] LongRanges { get; }
+
+        public string FacetField { get; set; } = "$facets";
```

Comment:
@nikcio should this also use `ExamineFieldNames.DefaultFacetsName`?

Reply:
Good catch. I've corrected it now 👍

Reply:
I actually thought I read somewhere that the FacetField wasn't used when using LongRange or DoubleRange facets. But I did some testing and it seems to work just fine. So I added methods to manipulate the facet field in the query to the Long and Double query fields too now.